libcontainer/cgroups/fs2/io.go

```diff
@@ -43,6 +45,16 @@ func setIo(dirPath string, r *configs.Resources) error {
 			}
 		}
 	}
+	for _, wd := range r.BlkioWeightDevice {
+		// per-device weight is supported since kernel v5.4, commit 795fe54c2a8.
+		if err := fscommon.WriteFile(dirPath, bfqWeight, wd.WeightString()); err != nil {
```

Comment:
This is unsafe to do unconditionally because old kernels called `kstrtoull` on the input string unconditionally -- and since `WeightString` is of the form `major:minor weight` we will end up setting the weight to the last `major` value in `BlkioWeightDevice`. We should check whether `bfq.weight_device` exists before writing these entries.